File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteRoomDatabase.java
package ro.danserboi.quotesformindandsoul;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Quote.class}, version = 1)
public abstract class QuoteRoomDatabase extends RoomDatabase {

    public abstract QuoteDao quoteDao();

    private static QuoteRoomDatabase INSTANCE;

    static QuoteRoomDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (QuoteRoomDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            QuoteRoomDatabase.class, "quote_database")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteRoomDatabase.kt
package ro.danserboi.quotesformindandsoul

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Quote::class], version = 1, exportSchema = true)
abstract class QuoteRoomDatabase : RoomDatabase() {
    abstract fun quoteDao(): QuoteDao?

    companion object {
        private var INSTANCE: QuoteRoomDatabase? = null
        @JvmStatic
        fun getDatabase(context: Context): QuoteRoomDatabase? {
            if (INSTANCE == null) {
                synchronized(QuoteRoomDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                QuoteRoomDatabase::class.java, "quote_database")
                                .fallbackToDestructiveMigration()
                                .build()
                    }
                }
            }
            return INSTANCE
        }
    }
}